File: app/src/androidTest/java/com/example/myapp/repository/TenantRepository.java
package com.example.olamigos.repository;

import com.example.olamigos.entity.Tenant;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface TenantRepository extends JpaRepository<Tenant, Long> {
    List<Tenant> findByNameContainingIgnoreCase(String name);
}

File: app/src/main/java/com/example/myapp/data/repository/TenantRepository.kt
package com.example.myapp.data.repository

import com.example.myapp.data.model.Tenant
import com.example.myapp.network.ApiClient

class TenantRepository {

    suspend fun createTenant(name: String): Tenant {
        return ApiClient.tenantApi.createTenant(Tenant(name = name))
    }

    suspend fun getAllTenants(): List<Tenant> {
        return ApiClient.tenantApi.getAllTenants()
    }

    suspend fun getPendingTenants(): List<Tenant> {
        return ApiClient.tenantApi.getPendingTenants()
    }

    suspend fun searchTenants(query: String): List<Tenant> {
        return ApiClient.tenantApi.searchTenants(query)
    }

    suspend fun approveTenant(id: Long): Tenant {
        return ApiClient.tenantApi.approveTenant(id)
    }
}
